File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int i = 0;
        int n = nums.size();
        int count = 0;
        int el = nums[i];
        while (i < n) {
            if (count == 0) {
                el = nums[i];
            }
            if (nums[i] == el) {
                count++;

            } else {
                count--;
            }
            i++;
        }

        int count2 = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == el) {
                count2++;
            }
            if (count2 > n / 2) {
                return el;
            }
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let i = 0;
    let j = nums.length
    let el = nums[0]
    let count = 0
    while (i < j) {
        if (count == 0) {
            el = nums[i]
        }
        if (nums[i] == el) {
            count++;
        } else {
            count--
        }
        i++;
    }
    let count2 = 0
    let n = nums.length
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == el) {
            count2++;
        }
        if (count2 > Math.floor(n / 2)) {
            return el
        }
    }

    return -1
};